src/azul/logging.py

```diff
@@ -163,20 +172,51 @@ def silenced_es_logger():
             assert es_log.level == original_log_level
 
 
-def http_body_log_message(body_type: str,
-                          body: bytes | bytearray | str | IO[bytes] | IO[str] | None,
+type BodyType = Literal['request', 'response']
+type PrintableBody = bytes | bytearray | str | Mapping
+type DigestableBody = PrintableBody | IO[bytes] | IO[str]
+
+http_body_log_prefix_len = 1024
+
+
+def http_body_log_message(body_type: BodyType,
+                          body: DigestableBody | None,
                           *,
-                          verbatim: bool = False,
+                          debug: int = 1,
                           ) -> str:
+    """
+    Auxiliary logging method assisting in logging request and response content
+    bodies.
+    """
     if body is None:
         return f'… without {body_type} body'
-    elif isinstance(body, (bytes, bytearray, str)):
-        if verbatim:
-            if isinstance(body, (bytes, bytearray)):
-                body = body.decode(errors='ignore')
+    elif debug > 0:
+        if isinstance(body, (bytes, bytearray, str, Mapping)):
```

Comment:
PL

Reply:
Opted for using JSON instead of Mapping.